lib/src/local_working_copy.rs

```diff
@@ -1768,7 +1840,7 @@ impl LockedWorkingCopy for LockedLocalWorkingCopy {
                 message: "Failed to read the working copy state".to_string(),
                 err: err.into(),
             })?;
-        self.tree_state_dirty |= tree_state.snapshot(options)?;
+        self.tree_state_dirty = self.tree_state_dirty.max(tree_state.snapshot(options)?);
```

Comment:
nit: can you add inline comment to `TreeStateChanged` type saying the order matters.

Alternatively, maybe we can use `bitflags`.